src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -19,34 +20,20 @@
 
 @Dao
 public class UserDaoJdbcImpl implements UserDao {
-    @Override
-    public Optional<User> findByLogin(String login) {
-        String query = "SELECT * FROM users WHERE login = ? AND user_deleted = FALSE; ";
-        User tempUser = new User();
-        try (Connection connection = ConnectionUtil.getConnection()) {
-            try (PreparedStatement preparedStatement = connection.prepareStatement(query)) {
-                preparedStatement.setString(1, login);
-                ResultSet resultSet = preparedStatement.executeQuery();
-                if (resultSet.next()) {
-                    tempUser = getUserFomResultSet(resultSet);
-                }
-            }
-        } catch (SQLException e) {
-            throw new DataProcessingException("Couldn't get user with login=" + login, e);
-        }
-        tempUser.setRoles(requestRoles(tempUser.getId()));
-        return Optional.of(tempUser);
-    }
 
     @Override
     public User create(User user) {
-        String query = "INSERT INTO users (user_name, login, user_password) VALUES (?, ?, ?);";
+        String query = "INSERT INTO users ("
+                + "user_name, login, user_password, salt) VALUES (?, ?, ?, ?);";
+        byte[] salt = HashUtil.getSalt();
```

Comment:
As I understand we need to implement hashing in Service - not on Dao level...

Reply:
Fixed